### PR TITLE
refactor(cli): minor refactor to helper functions and upgrade structopt 0.3.5

### PR DESCRIPTION
**Pull request by bochaco**
@dan-da this PR upgrades structopt, and I was also playing around with the parser but I couldn't get further than where you got to. So I did a minor refactor in the helper functions to read from stdin. I think we can keep this as is for now and see how we can keep evolving it once we try to implement more of reading args from stdin in other commands. 

One thing I noticed is that structopt shows a weird help message in terms of the arguments, it seems the `requires_if` causes that, see the ` <target> [target]` below:
```
$ safe files add -h
safe-files-add 0.6.0
Add a file to an existing FilesContainer on the network

USAGE:
    safe files add [FLAGS] [OPTIONS] <location> <target> [target]
...
```

### CODE REVIEW

File: safe-cli/subcommands/helpers.rs [hunk 1]
```diff
@@ -28,22 +28,27 @@ pub fn get_from_arg_or_stdin(
     message: Option<&str>,
 ) -> Result<String, String> {
     match target_arg {
-        Some(ref t) if t == "" => get_from_stdin(message),
+        Some(ref t) if t.is_empty() => {
+            let val = get_from_stdin(message)?;
+            Ok(String::from_utf8_lossy(&val).to_string())
```
Comment (dan-da) [inline safe-cli/subcommands/helpers.rs]: I worry about using from_utf8_lossy(), as it silently alters the user's input if any non utf8 chars encountered.  User may be using a locale other than UTF-8, or paste in a URL with invalid chars obtained from elsewhere.    I would think its better to use the non lossy version and throw an error.

File: safe-cli/subcommands/helpers.rs [hunk 2]
```diff
@@ -28,22 +28,27 @@ pub fn get_from_arg_or_stdin(
...
+        }
         Some(t) => Ok(t),
-        None => get_from_stdin(message),
+        None => {
+            let val = get_from_stdin(message)?;
+            Ok(String::from_utf8_lossy(&val).to_string())
```
Comment (dan-da) [inline safe-cli/subcommands/helpers.rs]: here also

**Review by dan-da — CHANGES_REQUESTED** [review]
Looks good overall, just maybe we should not use from_utf8_lossy().   what do you think?

**Comment by bochaco** [comment]
Agreed @dan-da , I just pushed a commit to fix that.

**Pull request merged**